dali/operators/generic/slice/slice_attr.h

```diff
@@ -199,6 +185,86 @@ class SliceAttr {
       };
   }
 
+ private:
+  const OpSpec &spec_;
+
+  std::vector<int> axes_;
+  TensorLayout axis_names_;
+
+  ArgValue<int, 1> start_;
+  ArgValue<float, 1> rel_start_;
+
+  ArgValue<int, 1> end_;
+  ArgValue<float, 1> rel_end_;
+
+  ArgValue<int, 1> shape_;
+  ArgValue<float, 1> rel_shape_;
+
+  std::vector<CropWindowGenerator> crop_window_generators_;
+
+  bool has_start_, has_end_, has_shape_;
+};
+
+
+class PositionalSliceAttr {
+ public:
+  explicit inline PositionalSliceAttr(const OpSpec& spec) : spec_(spec) {
+    normalized_anchor_ = spec.GetArgument<bool>("normalized_anchor");
+    normalized_shape_ = spec.GetArgument<bool>("normalized_shape");
+    int max_batch_sz = spec.GetArgument<int>("max_batch_size");
+    crop_window_generators_.resize(max_batch_sz);
+    const bool has_axes_arg = spec.HasArgument("axes");
+    const bool has_axis_names_arg = spec.HasArgument("axis_names");
+    // Process `axis_names` if provided, or if neither `axis_names` nor `axes` are
+    if (has_axis_names_arg || !has_axes_arg) {
```

Comment:
What if both "axes" and "axis_names" are provided?

Reply:
The new logic should be clearer